src/main/java/net/neoforged/neoforge/resource/ResourcePackLoader.java

```diff
@@ -100,7 +101,7 @@ private static void packFinder(Map<IModFile, Pack.ResourcesSupplier> modResource
                         name,
                         Component.literal(packName.isEmpty() ? "[unnamed]" : packName),
                         PackSource.DEFAULT,
-                        Optional.empty());
+                        Optional.of(new KnownPack("neoforge", name, mod.getVersion().toString())));
```

Comment:
Could you briefly explain why the namespace is `neoforge`, rather than the mod ID (and the pack ID being the pack name)?

Reply:
Because neoforge is the one providing the pack, not the mod -- all the logic for the pack discovery and creation comes from neoforge. We'd be placing something into a mod's namespace without it doing anything if we did it the other way.